From 75e6e28f4720a0d60d5996e55794409bf404e513 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Mon, 31 Aug 2009 15:13:29 +0800
Subject: [PATCH 074/256] ENGR00116086 ipu lib: add FORCE flag to set fb var

Sometime, if the fb_var is the same as default, it will not
be set, add FORCE flag to avoid this issue.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index 7ed5895..a712ce2 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -974,6 +974,7 @@ static int _ipu_mem_alloc(ipu_lib_input_param_t * input,
 				fb_var.xres_virtual = fb_var.xres;
 				fb_var.yres = oheight;
 				fb_var.yres_virtual = fb_var.yres * 2;
+				fb_var.activate |= FB_ACTIVATE_FORCE;
 				if ( ioctl(ipu_priv_handle->output[j].fd_fb, FBIOPUT_VSCREENINFO, &fb_var) < 0) {
 					dbg(DBG_ERR, "Set FB var info failed!\n");
 					close(ipu_priv_handle->output[j].fd_fb);
-- 
1.8.0

